### PR TITLE
Asking for Help on Implementing K_SINK in Transformation

### PR DESCRIPTION
**Pull request by JeffNeff**
I know this is not pretty right now.. but why is it not working? 

I cant get `K_SINK` to assign to anything. It always displays as a `nil` value. 

Testing with this yaml
```
apiVersion: serving.knative.dev/v1
kind: Service
metadata:
    name: event-display
spec:
    template:
        spec:
            containers:
                -
                    image: >-
                        gcr.io/knative-releases/knative.dev/eventing-contrib/cmd/event_display@sha256:46d5a66f300c3ced590835d379a0e9badf413ae7ab60f21a2550ecedbc9eb9d3


---

apiVersion: flow.triggermesh.io/v1alpha1
kind: XMLToJSONTransformation
metadata:
  name: demo
spec:
  sink:
    ref:
      apiVersion: serving.knative.dev/v1
      kind: Service
      name: event-display

---

apiVersion: sources.knative.dev/v1
kind: PingSource
metadata:
  name: pingxml
spec:
  schedule: "*/1 * * * *"
  contentType: application/xml
  data: '<note><to>Tove</to><from>Jani</from><heading>Reminder</heading><body>Dont forget me this weekend</body></note>'
  sink:
    ref:
      apiVersion: flow.triggermesh.io/v1alpha1
      kind: XMLToJSONTransformation
      name: demo
```

### CODE REVIEW

**Comment by JeffNeff** [comment]
kubectl -n test describe xmltojsontransformation/demo 

```
...
Spec:
  Sink:
    Ref:
      API Version:  serving.knative.dev/v1
      Kind:         Service
      Name:         event-display
Status:
  Conditions:
    Last Transition Time:  2022-01-19T19:00:27Z
    Message:               unsuccessfully observed a new generation
    Reason:                NewObservedGenFailure
    Status:                Unknown
    Type:                  Ready
  Observed Generation:     1
Events:
  Type     Reason        Age                From                                Message
  ----     ------        ----               ----                                -------
  Warning  SinkResolver  44s (x2 over 44s)  xmltojsontransformation-controller  destination missing Ref and URI, expected at least one
```

**Comment by antoineco** [comment]
The sink's URL needs to be resolved by the controller, then passed to the adapter constructor somehow.

An example of resolving the URL:

https://github.com/triggermesh/triggermesh/blob/60d2bcda6835921bebce0278bc41cb0f109bb36a/pkg/sources/reconciler/common/reconcile.go#L233

The Filter and Splitter components do something similar.

**Comment by JeffNeff** [comment]
> The sink's URL needs to be resolved by the controller, then passed to the adapter constructor somehow.
> 
> An example of resolving the URL:
> 
> https://github.com/triggermesh/triggermesh/blob/60d2bcda6835921bebce0278bc41cb0f109bb36a/pkg/sources/reconciler/common/reconcile.go#L233
> 
> The Filter and Splitter components do something similar.

I thought I was doing that already on line 48 of `pkg/flow/reconciler/xmltojsontransformation/reconciler.go` ?

File: pkg/flow/reconciler/xmltojsontransformation/reconciler.go [hunk 1]
```diff
@@ -40,7 +45,13 @@ var _ reconcilerv1alpha1.Interface = (*Reconciler)(nil)
 
 // ReconcileKind implements Interface.ReconcileKind.
 func (r *Reconciler) ReconcileKind(ctx context.Context, trg *v1alpha1.XMLToJSONTransformation) pkgreconciler.Event {
-	adapter, event := r.ksvcr.ReconcileKService(ctx, trg, makeTargetAdapterKService(trg, r.adapterCfg))
+	sink, err := r.base.SinkResolver.URIFromDestinationV1(ctx, trg.Spec.Sink, trg)
```
Comment (JeffNeff) [inline pkg/flow/reconciler/xmltojsontransformation/reconciler.go:48]: here @antoineco

**Comment by JeffNeff** [comment]
I am just going to

**Pull request closed without merging**